pkg/ccl/backupccl/backup_planning.go

```diff
@@ -699,16 +700,22 @@ func backupPlanHook(
 		if backupStmt.Nested {
 			if backupStmt.AppendToLatest {
 				initialDetails.Destination.Subdir = latestFileName
+				initialDetails.Destination.Exists = true
+
 			} else if subdir != "" {
 				initialDetails.Destination.Subdir = "/" + strings.TrimPrefix(subdir, "/")
```

Comment:
since we can't buffer a warning during execution (i.e. in resolveDest [here](https://github.com/cockroachdb/cockroach/pull/79799/commits/69259ec995c1db0758cba8375f069facdead72c3#r847854846)), I propose reading from external storage here, just see if the subdirectory exists. I realize this breaks the general rule of waiting until execution to read from external storage.

Reply:
I think I'd rather keep the rule about fast planning and io in exec, and just throw the error during exec (maybe gated on a setting) in 22.1 if we can't buffer a notice. Separately we should figure out a way to accumulate notices in job exec and then send them to the client if it is still connected, and display them on the job status after the fact, in 22.2.

Reply:
Done. Added a cluster setting.

Reply:
should we put "deprecated" somewhere in the setting name since we intend to take this away entirely in 22.2?